### PR TITLE
Refactoring 

### PR DESCRIPTION
**Pull request by loganb294**
## Refactor Using Repository Pattern

The repository pattern is a design pattern that separates data access logic from business logic, which is widely adopted in modern web applications. This approach provides several key benefits:

### Benefits
- **Separation of Concerns:** Clearly distinguishes data access logic from business logic, making the codebase easier to maintain and test.
- **Testability:** Facilitates unit testing by allowing the data access logic and business logic to be tested independently without requiring a database setup.
- **Flexibility:** Simplifies switching between different data access technologies (e.g., from SQL to NoSQL) without altering the business logic.

---

## Update Swagger Documentation

To reflect the API changes made during the refactor, I will update the Swagger documentation, particularly focusing on the AUTH routes. This update aims to enable user creation through Swagger, similar to the login page functionality.

### Benefits
- **Enhanced API Usability:** Enables users to create new accounts directly through the Swagger UI, improving the development and testing experience.
- **Consistent Documentation:** Ensures that Swagger documentation accurately represents the new repository pattern and updated API routes.
- **Improved Testing:** Provides a more effective way to test API endpoints, including user creation and authentication, via Swagger.

---

## Implementation Plan

The refactor will involve creating a **repository folder** and updating the relevant files in the **controller** and **routes** folders. Specifically, I will work on the following:

- **userRepository:** Implement the repository pattern in the `repository` folder to handle user data access.
- **userController:** Update the `userController` in the `controller` folder to interact with the new repository layer.
- **baseController:** Modify the `baseController` in the `controller` folder as needed to align with the new repository pattern.
- **userRoutes:** Revise the `userRoutes` in the `routes` folder to reflect changes in the controller methods and repository integration.

---

## linked issues
- [refactor](https://github.com/BIT-Studio-3/group-project-24-2-s-m-l-l/issues/122)

### CODE REVIEW

**Comment by loganb294** [comment]
@Samuel1Bridges you are the main reviewer 
@MikeBike2 you are secondary reviewer

**Review by Samuel1Bridges — APPROVED** [review]
Does it fit the purpose:
Yes it fits the purpose.

Readable, consistent and structured?:
The code is readable and structured.

Use consistent naming conventions:
yes
Are there sufficient and useful comments?:
There could be a few more comments in the controller's.

Don't Repeat Yourself:
it doesn't repeat its self

Does the code function correctly?:
Yes

Is the code reliable?:
yes
Has it solved the stated issue?:
yes
Tested?:
yes
Performance or speed issues?:
not checked
Scalable?:
yes
Compatible across all platforms?:
not checked

File: controllers/v1/baseController.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      message: `Entity with the id: ${req.params.id} successfully deleted`,
+    });
+  } catch (err) {
+    return res.status(STATUS_CODES.SERVER_ERROR).json({
+      message: err.message,
+    });
+  }
+};
+
+export { createEntity, getEntities, getEmailResource, getEntity, updateEntity, deleteEntity };
```
Comment (MikeBike2) [inline controllers/v1/baseController.js:128]: thanks for giving me a heads up earlier about this change, forgot to implement it into my code but can just do it on the merge i think !

Comment (loganb294) [reply controllers/v1/baseController.js:128]: if you need a hang just ask

**Review by MikeBike2 — APPROVED** [review]
Does it fit the purpose:
Yes it fits the purpose.
Readable, consistent and structured?:
The code is readable and structured.

Use consistent naming conventions:
yes
Are there sufficient and useful comments?:
yes improved commenting since last pull request 

Don't Repeat Yourself:
it doesn't repeat itsself

Does the code function correctly?:
Yes

Is the code reliable?:
yes
Has it solved the stated issue?:
yes
Tested?:
yes
Performance or speed issues?:
not checked
Scalable?:
yes
Compatible across all platforms?:
not checked
might have to make some changes with my code on merge but was informed of this decision on name changes, we can do this on the day when we merge

**Pull request merged**